operator/pkg/lib/sidecars/proxy.go

```diff
@@ -0,0 +1,49 @@
+package sidecars
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"github.com/kyma-project/istio/operator/pkg/lib/sidecars/pods"
+	"github.com/kyma-project/istio/operator/pkg/lib/sidecars/restart"
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func ProxyReset(ctx context.Context, c client.Client, expectedImage pods.SidecarImage, namespaceEnabledByDefault, cniEnabled bool, logger *logr.Logger) ([]restart.RestartWarning, error) {
+	differentImagePodList, err := pods.GetPodsWithDifferentSidecarImage(ctx, c, expectedImage)
+	if err != nil {
+		return nil, err
+	}
+
+	noSidecarPodList, err := pods.GetPodsWithoutSidecar(ctx, c, namespaceEnabledByDefault)
```

Comment:
Looks to me as though we are missing a negative case (pods that have sidecar right now but shouldn't)

Reply:
yes, so far we have supported only pods that should had a sidecar but they didn't

Reply:
If it was not relevant in the past, do we have evidence that this is a valid case to support in the future?

Reply:
I was sure that we are supporting this in reconciler :O
We should discuss this functionality then

Reply:
It was mainly implemented to fix the issue of missing sidecar injection(mutating webhook), meaning that normally if you change label/annotation to disable, it will get restarted automatically.
With the default strategy of sidecar injection being opt-in instead of opt-out, it doesn't feel crucial to support this the same way.

But I see one case that we can extend, when you have a deployment with sidecar injected and you change its namespace label to `istio-injection=disabled`.

Reply:
I think we should add a new issue for this and discuss it in the refinement and then decide if we want to implement it or not.

Reply:
I will adapt the component test for now and leave a note in case we want to extend that

Reply:
https://github.com/kyma-project/istio/pull/37/commits/10d33020c88abfdbc2ac347f06250f0d8f5574da

Reply:
As discussed in refinement removed support for missing sidecars in https://github.com/kyma-project/istio/pull/37/commits/50d9428e802a2b8778891ac96c68aea917c5de6b